05_concurrency/homework/homework.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Item struct {
+	value interface{}
+	ttl   time.Duration
+}
+
+type CacheBuffer struct {
+	sync.Mutex
+	items    map[string]*Item
+	cacheTTL map[string]time.Duration //
+}
+
+//worker sets off timer before key invalidation, after key is invalid it gets deleted and its ttl from ttl map
+func (cb *CacheBuffer) worker(key string) {
+	cb.Lock()
+	ttl := cb.items[key].ttl
+	cb.Unlock()
+	<-time.After(ttl)
+	cb.Lock()
+	delete(cb.items, key)
+	delete(cb.cacheTTL, key)
+	cb.Unlock()
+	fmt.Println("Deleted item with key: ", key)
+}
+
+func (cb *CacheBuffer) Set(key string, value interface{}, ttl time.Duration) {
+	cb.Lock()
+	cb.items[key] = &Item{value: value, ttl: ttl}
+	cb.cacheTTL[key] = ttl
+	go cb.worker(key)
```

Comment:
if line 22 will execute before line 38 - you will lock worker forever. Move please 37 line after `cb.Unlock()`